File: backend/app/schemas/rbac.py
from pydantic import BaseModel
from typing import List, Optional
from uuid import UUID

from backend.app.models import RoleScope

class PermissionBase(BaseModel):
    name: str  # e.g., "attendance:write", "teacher:read"
    description: Optional[str] = None

class PermissionCreate(PermissionBase):
    pass

class PermissionResponse(PermissionBase):
    id: UUID

class RoleBase(BaseModel):
    name: str  # e.g., "admin", "class_teacher", "principal"
    description: Optional[str] = None

class RoleCreate(RoleBase):
    pass


class RoleUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None


class RoleResponse(RoleBase):
    id: UUID
    scope: RoleScope
    school_id: Optional[UUID] = None

class RoleWithPermissionsResponse(RoleResponse):
    permissions: List[PermissionResponse]

class AssignPermissionsRequest(BaseModel):
    permission_ids: List[UUID]

File: frontend/src/types/rbac.ts
export interface PermissionResponse {
  id: string;
  name: string;
  description: string | null;
}

export type RoleScope = "PLATFORM" | "SCHOOL";

export interface RoleResponse {
  id: string;
  name: string;
  description: string | null;
  scope: RoleScope;
  school_id: string | null;
}

export interface RoleWithPermissionsResponse extends RoleResponse {
  permissions: PermissionResponse[];
}
